include/aws/s3/private/s3_default_meta_request.h

```diff
@@ -18,7 +18,7 @@ struct aws_s3_meta_request_default {
     /* Actual type for the single request (may be AWS_S3_REQUEST_TYPE_UNKNOWN) */
     enum aws_s3_request_type request_type;
 
-    /* S3 operation name for the single request (NULL if unknown) */
+    /* S3 operation name for the single request */
```

Comment:
should we warn people that if they don't set it right, they may still have the issue downloading the error xml?

Or some documentation about what will happen if we don't know the operation name.

Reply:
I added very explicit details to the docs in s3_client.h